projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, EventEmitter, HostBinding, Input, Output } from '@angular/core';
+
+/**
+ * Interface for visibility item data
+ */
+export interface VisibilityItem {
+
+  /**
+   * Id of the item
+   */
+  id: number;
+
+  /**
+   * Name of the item
+   */
+  name: string;
+
+  /**
+   * Whether the item is currently highlighted
+   */
+  visible: boolean;
+
+  /**
+   * Src of the icon (visible or non-visible)
+   */
+  iconSrc: string;
```

Comment:
Why is this even in here? Can't you just switch the icon based on the visible property in the component?

Reply:
Not sure, it's something I had in there from the beginning. I agree it doesn't make much sense to do it this way though.

Reply:
See https://github.com/hubmapconsortium/ccf-ui/pull/337/files#diff-929a0ad388df6a0e8cdabed57f704b91R2 in the same PR for how Adam did it using just the visible property.

Reply:
👍